app/src/main/java/org/schabi/newpipe/settings/tabs/Tab.java

```diff
@@ -482,4 +489,128 @@ private String getDefaultKioskId(final Context context) {
+            this.playlistName = playlistName;
+            this.playlistType = LocalItemType.PLAYLIST_REMOTE_ITEM;
+            this.playlistId = -1;
+        }
+
+        public PlaylistTab(final JsonObject jsonObject) {
+            super(jsonObject);
+        }
+
+        @Override
+        public int getTabId() {
+            return ID;
+        }
+
+        @Override
+        public String getTabName(final Context context) {
+            return playlistName;
+        }
+
+        @DrawableRes
+        @Override
+        public int getTabIconRes(final Context context) {
+            return ThemeHelper.resolveResourceIdFromAttr(context, R.attr.ic_bookmark);
+        }
+
+        @Override
+        public Fragment getFragment(final Context context) {
+            if (playlistType == LocalItemType.PLAYLIST_LOCAL_ITEM) {
+                return LocalPlaylistFragment.getInstance(playlistId, playlistName);
+
+            } else { // playlistType == LocalItemType.PLAYLIST_REMOTE_ITEM
+                return PlaylistFragment.getInstance(playlistServiceId, playlistUrl, playlistName);
+            }
+        }
+
+        @Override
+        protected void writeDataToJson(final JsonSink writerSink) {
+            writerSink.value(JSON_PLAYLIST_SERVICE_ID_KEY, playlistServiceId)
+                    .value(JSON_PLAYLIST_URL_KEY, playlistUrl)
+                    .value(JSON_PLAYLIST_NAME_KEY, playlistName)
+                    .value(JSON_PLAYLIST_ID_KEY, playlistId)
+                    .value(JSON_PLAYLIST_TYPE_KEY, playlistType.toString());
+        }
+
+        @Override
+        protected void readDataFromJson(final JsonObject jsonObject) {
+            playlistServiceId = jsonObject.getInt(JSON_PLAYLIST_SERVICE_ID_KEY, -1);
+            playlistUrl = jsonObject.getString(JSON_PLAYLIST_URL_KEY, "<no-url>");
+            playlistName = jsonObject.getString(JSON_PLAYLIST_NAME_KEY, "<no-name>");
+            playlistId = jsonObject.getInt(JSON_PLAYLIST_ID_KEY, -1);
+            playlistType = LocalItemType.valueOf(
+                    jsonObject.getString(JSON_PLAYLIST_TYPE_KEY,
+                            LocalItemType.PLAYLIST_LOCAL_ITEM.toString())
+            );
+        }
+
+        @Override
+        public boolean equals(final Object obj) {
+            final boolean baseEquals = super.equals(obj)
+                    && Objects.equals(playlistType, ((PlaylistTab) obj).playlistType)
+                    && Objects.equals(playlistName, ((PlaylistTab) obj).playlistName);
+
+            if (!baseEquals) {
+                return false;
+            }
+
+            final boolean localPlaylistEquals = playlistId == ((PlaylistTab) obj).playlistId;
+            final boolean remotePlaylistEquals =
+                    playlistServiceId == ((PlaylistTab) obj).playlistServiceId
+                            && Objects.equals(playlistUrl, ((PlaylistTab) obj).playlistUrl);
+
+            return localPlaylistEquals || remotePlaylistEquals;
```

Comment:
This allows the `remotePlaylistEquals` part to not be calculated when `localPlaylistEquals` is `true` and also looks better imo
```suggestion
            return playlistId == ((PlaylistTab) obj).playlistId                               // local
                    || (playlistServiceId == ((PlaylistTab) obj).playlistServiceId            // remote
                            && Objects.equals(playlistUrl, ((PlaylistTab) obj).playlistUrl));
```